src/main/java/org/generationcp/middleware/service/impl/study/StudyServiceImpl.java

```diff
@@ -520,6 +524,81 @@ public StudyDetailsDto getStudyDetailsForGeolocation(final Integer geolocationId
 		}
 	}
 
+	@Override
+	public StudyDetailsDto getStudyDetailsByGeolocation(final Integer geolocationId) {
+		try {
+			final StudyMetadata studyMetadata = this.studyDataManager.getStudyMetadataForGeolocationId(geolocationId);
+			if (studyMetadata != null) {
+				final StudyDetailsDto studyDetailsDto = new StudyDetailsDto();
+				studyDetailsDto.setMetadata(studyMetadata);
+
+				final List<UserDto> users = new ArrayList<>();
+				users.addAll(this.studyDataManager.getUsersForEnvironment(studyMetadata.getStudyDbId()));
+				users.addAll(this.studyDataManager.getUsersAssociatedToStudy(studyMetadata.getNurseryOrTrialId()));
+				studyDetailsDto.setContacts(users);
+
+				final DmsProject environmentDataset =
+					this.daoFactory.getDmsProjectDAO().getDatasetsByTypeForStudy(studyMetadata.getTrialDbId(), DatasetTypeEnum.SUMMARY_DATA.getId()).get(0);
+				final List<MeasurementVariable> environmentConditions = this.daoFactory.getDmsProjectDAO()
+					.getObservationSetVariables(environmentDataset.getProjectId(), Lists.<Integer>newArrayList(VariableType.STUDY_CONDITION.getId()));
+				List<Integer> variableIds = environmentConditions.stream().map(MeasurementVariable::getTermId).collect(Collectors.toList());
+				final List<MeasurementVariable> environmentParameters = new ArrayList<>();
+				environmentParameters.addAll(this.studyDataManager.getEnvironmentConditionVariablesByGeoLocationIdAndVariableIds(geolocationId, variableIds));
+
+				final List<MeasurementVariable> environmentDetails = this.daoFactory.getDmsProjectDAO()
+					.getObservationSetVariables(environmentDataset.getProjectId(), Lists.<Integer>newArrayList(VariableType.ENVIRONMENT_DETAIL.getId()));
+				variableIds = environmentDetails.stream().map(MeasurementVariable::getTermId).collect(Collectors.toList());
+				environmentParameters.addAll(this.studyDataManager.getEnvironmentDetailVariablesByGeoLocationIdAndVariableIds(geolocationId, variableIds));
+
+				final List<MeasurementVariable> environmentVariables = new ArrayList<>(environmentConditions);
+				environmentVariables.addAll(environmentDetails);
+				environmentParameters.addAll(createGeolocationVariables(environmentVariables, geolocationId));
+				studyDetailsDto.setEnvironmentParameters(environmentParameters);
+
+				final Map<String, String> properties = new HashMap<>();
+				variableIds = environmentVariables.stream().map(MeasurementVariable::getTermId).collect(Collectors.toList());
+				properties.putAll(this.studyDataManager.getGeolocationPropsAndValuesByGeolocation(geolocationId, variableIds));
```

Comment:
I don't think this line is necessary:
`properties.putAll(this.studyDataManager.getGeolocationPropsAndValuesByGeolocation(geolocationId, variableIds));`
I'm thinking we can just keep this Middleware method that sets environment parameters then in BMSAPI we can "move" the environment parameters to `additionalProperties` for the v1 call,  what do you think?